zcash_client_sqlite/src/wallet/init/migrations/shardtree_support.rs

```diff
@@ -99,7 +99,8 @@ impl RusqliteMigration for Migration {
                 checkpoint_id INTEGER NOT NULL,
                 mark_removed_position INTEGER NOT NULL,
                 FOREIGN KEY (checkpoint_id) REFERENCES sapling_tree_checkpoints(checkpoint_id)
-                ON DELETE CASCADE
+                ON DELETE CASCADE,
+                CONSTRAINT spend_position_unique UNIQUE (checkpoint_id, mark_removed_position)
```

Comment:
For test wallets, not having this constraint isn't a problem; I just added it because it's semantically correct and if it gets violated it means we have a bug somewhere.